packages/deploy/src/stateTransform.ts

```diff
@@ -472,7 +472,8 @@ function idKeyPairs<P extends { id: string }, S extends { id: string }>(
   stateItems: Record<keyof S, S>
 ): [key: string, projectItem: P, stateItem: S][] {
   let pairs: [string, P, S][] = [];
-  for (const projectItem of projectItems) {
+  for (let i = 0; i < projectItems.length; i++) {
```

Comment:
This is a curious diff?

Does for/of cause something to fail in node 22 here?

Reply:
Changed it because not all list data passed to the `idKeyPairs` function is an `iterable` (had some errors on that). might be an array but not an `iterable`. Hence the `for..of` syntax will error but our classic looping works on both `iterables` and `non-iterable` lists.

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols